### PR TITLE
test(hansbug): add benchmark for the basic operations and functions

### PR DESCRIPTION
**Pull request by HansBug**
## Description

Add benchmark for 

* [x] class `TreeStorage` and function `raw`
* [x] class `TreeValue`
* [x] utility functions of `TreeValue`
* [x] auto operations in `FastTreeValue`
* [x] Comparation with `dm-tree` : https://github.com/deepmind/tree
* [x] Comparation with `tianshou-Batch` : https://tianshou.readthedocs.io/en/master/tutorials/batch.html

## Check List

- [x] merge the latest version source branch/repo, and resolve all the conflicts
- [x] pass style check
- [x] pass all the tests

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -8,7 +8,10 @@ pytest-forked~=1.3.0
 pytest-xdist~=1.34.0
 pytest-timeout~=1.4.2
 pytest-rerunfailures~=9.1.1
+pytest-benchmark~=3.4.0
 numpy>=1.19,<2
 torch>=1.3.1,<=1.8.0
 easydict>=1.7,<2
 testtools>=2
+dm-tree>=0.1.6
+tianshou>=0.4.0
```
Comment (PaParaZz1) [inline requirements-test.txt:17]: use the latest tianshou version (0.4.5)

File: test/compare/test_dm_tree.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import pytest
+import tree
```
Comment (PaParaZz1) [inline test/compare/test_dm_tree.py:2]: add comments for preventing confusing, such as:
```python
import tree  # dm-tree https://github.com/deepmind/tree
```

File: test/compare/test_tianshou_batch.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+_TREE_DATA_1 = {'a': 1, 'b': 2, 'x': {'c': 3, 'd': 4}}
+_TREE_1 = FastTreeValue(_TREE_DATA_1)
+_BATCH_1 = Batch(**_TREE_DATA_1)
+
+_TREE_DATA_2 = {'a': torch.randn(2, 3), 'x': {'c': torch.randn(3, 4)}}
+_TREE_2 = FastTreeValue(_TREE_DATA_2)
+_BATCH_2 = Batch(**_TREE_DATA_2)
+
+_TREE_DATA_3 = {'a': torch.randn(30, 40), 'x': {'c': torch.randn(30, 50)}}
```
Comment (PaParaZz1) [inline test/compare/test_tianshou_batch.py:15]: use real data shape

**Review by PaParaZz1 — APPROVED** [review]

File: requirements-test.txt (hunk 1 above)
Comment (HansBug) [reply requirements-test.txt:17]: Fixed on `0.4.5`, or just `>=0.4.5` ?

File: test/compare/test_dm_tree.py (hunk 2 above)
Comment (HansBug) [reply test/compare/test_dm_tree.py]: Completed.

File: test/compare/test_tianshou_batch.py (hunk 3 above)
Comment (HansBug) [reply test/compare/test_tianshou_batch.py]: Completed.

File: requirements-test.txt (hunk 1 above)
Comment (HansBug) [reply requirements-test.txt]: Completed.

**Pull request merged**